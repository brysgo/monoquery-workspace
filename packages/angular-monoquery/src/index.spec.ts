import { HttpClientModule } from "@angular/common/http";
import { ApolloModule, Apollo } from "apollo-angular";
import { HttpLinkModule, HttpLink } from "apollo-angular-link-http";

import { TestBed } from "@angular/core/testing";

import gql from "graphql-tag";

import { MonoQueryModule } from "./";

describe("MonoQueryModule", () => {
  it("does things", async () => {
    const FakeModule = { fragments: { aFragment: `` } };
    const FakeMonoProvider = MonoQueryModule.forChild(FakeModule);
    TestBed.configureTestingModule({
      imports: [
        HttpClientModule,
        ApolloModule,
        HttpLinkModule,
        MonoQueryModule.forRoot()
      ],
      providers: [
        HttpClientModule,
        ApolloModule,
        HttpLinkModule,
        FakeMonoProvider
      ]
    });
    fakeMonoProvider = TestBed.get(FakeMonoProvider);
    expect(await fakeMonoProvider).toMatchSnapshot();
  });
});
