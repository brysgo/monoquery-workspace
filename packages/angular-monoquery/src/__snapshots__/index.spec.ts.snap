// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MonoQueryModule is used to build a MonoProvider 1`] = `
<example-component
  jsonData={[Function Observable]}
  monoProvider={[Function MonoProvider]}
>
  {"simpleFragment":{"hello":"world"}}
</example-component>
`;

exports[`MonoQueryModule works when the fetcher returns a promise 1`] = `
<example-component
  jsonData={[Function Observable]}
  monoProvider={[Function MonoProvider]}
>
  {"simpleFragment":{"hello":"world"}}
</example-component>
`;

exports[`MonoQueryModule works when the fetcher returns an observer 1`] = `
<example-component
  jsonData={[Function Observable]}
  monoProvider={[Function MonoProvider]}
>
  {"simpleFragment":{"hello":"world"}}
</example-component>
`;

exports[`MonoQueryModule works when the fetcher returns an observer 2`] = `
<example-component
  jsonData={[Function Observable]}
  monoProvider={[Function MonoProvider]}
>
  {"simpleFragment":{"hello":"again"}}
</example-component>
`;
